app/lib/submission_builder/ty2022/states/authentication_header.rb

```diff
@@ -75,6 +51,35 @@ def document
             end
           end
         end
+
+        def xml_type_for_state_id(state_id)
+          if state_id&.id_type_driver_license?
+            "DrvrLcns"
+          elsif state_id&.id_type_dmv_bmv?
+            "StateIssdId"
+          end
+        end
+
+        def state_id_to_xml(state_id, xml_builder)
+          xml_type = xml_type_for_state_id(state_id)
+          if xml_type
+            xml_builder.send("#{xml_type}Num", state_id.id_number)
+            xml_builder.send("#{xml_type}StCd", state_id.state)
+            if state_id.first_three_doc_num.present?
+              xml_builder.send("#{xml_type}AddInfo", state_id.first_three_doc_num)
```

Comment:
Does the frontend only ask for the additional first three characters if issued in New York? I think I saw in the story that this was an issue.

Reply:
Correct. The frontend behaves this way - only New York IDs have this additional info